Share to socials button

alexmassen-hane opened the pull request:
Made a popover button with links to Twitter, Facebook and LinkedIn so that it will share the current institution or country page. 

Have to clean it up, get the links working for each social and change colour of the buttons to match existing theme.
Also added a "copy link" button that will copy the link to the current page that you are on. 

![image](https://user-images.githubusercontent.com/104514872/189320126-fa58fcb4-e49e-4b47-ad6d-6c84b4b11ded.png)

jdddog reviewed (changes requested):
Hey Alex, it looks like good progress, here is some feedback on the appearance, I'll give some feedback on the code in the next review.

* The share button should look similar to the Wikipedia and Download buttons. You could use the FiShare2 React Icon: https://react-icons.github.io/react-icons/search?q=fishare2
* So that we can add more sharing options in a scalable way, it would be good to use a two column grid, with an icon and label underneath (see picture below). Then we can also add link copying and embedding.
* For the link copying, it would be good to show a toast when the link has been copied to for feedback to the user, the toast should disappear by itself: https://chakra-ui.com/docs/components/toast
* Remove the blue outline from the popup.
* We should figure out where to show it on mobile, but that can be a next step.

Our own icons via IcoMoon and React Icons might cause problems. We might need to add a share and link icon in IcoMoon so the code can stay consistent.

See here for an example sketch:
![image](https://user-images.githubusercontent.com/5715104/194795159-47a34780-66bd-4f5a-973a-0565d8fb873f.png)

alexmassen-hane commented:
- Changed style of the button and the popover window.
- Added toast popup when clicking on the "Link" button and tells the user it has been copied. 
- Added the embed button back but need to add functionality to it.
- Need to add share button to mobile version of the site. Only for desktop for now.
- Need to add unit tests for the share button and the links that it goes to. 

![image](https://user-images.githubusercontent.com/104514872/197155549-a728a334-4070-40dc-89aa-429c919a24f6.png)

alexmassen-hane commented:
- Commented out embed icon as it's not being used yet.
- Social share links open in new tab. 
- Aligned social share icons.
- Changed toast to be at the top of the page and changed it to the brand colour. 
- Made a mobile version for the share button.
- Added appropriate unit tests. 

![image](https://user-images.githubusercontent.com/104514872/198234342-a6f58e33-e728-42f7-a6ff-4509fe1df4bf.png)

![Screen Shot 2022-10-27 at 4 37 49 pm](https://user-images.githubusercontent.com/104514872/198234883-0fda5dc8-8c3a-404a-a5ff-f61f606b1807.png)

jdddog commented on components/details/SharebuttonLinks.tsx:
```diff
@@ -0,0 +1,177 @@
+  const { hasCopied, onCopy } = useClipboard(value);
+
+  return (
+    <Popover>
```
It would be good to move the popover styling into a theme.

jdddog commented on components/details/SharebuttonLinks.tsx:
```diff
@@ -0,0 +1,177 @@
+        </Button>
+      </PopoverTrigger>
+      <Portal>
+        <PopoverContent
```
So that the popover doesn't overlap the right side of the window, you can add PopoverContent mr={"18px"}

jdddog commented on components/details/SharebuttonLinks.tsx:
```diff
@@ -0,0 +1,177 @@
+          data-test={`${platform} share popover panel`}
+          width="min-content"
+          height="min-content"
+          _focus={{ border: "none" }}
```
If you compare the popover to the default popover, the popover jiggles a bit and doesn't animate smoothly. I think it is because of the _focus={{ border: "none" }} tag, you might need to delve a little deeper into the theme to see how to turn off the blue outer border without disrupting the animation and creating the jiggling.

jdddog commented on components/details/SharebuttonLinks.tsx:
```diff
@@ -0,0 +1,177 @@
+const CopyLink = () => {
+  const toast = useToast();
+  React.useEffect(() => {
+    toast({
```
It would be good if for the link copy toast:
- The text read: "Copied link: https://open.coki.ac/country/AUS"
- On mobile: make the toast show up in the middle of the header.

You might be able to not set the width, as the text will fill up the toast. Good to check how it will behave on mobile when the toast has the whole URL in it.

jdddog commented on package.json:
```diff
@@ -21,6 +21,7 @@
     "@nivo/pie": "^0.79.1",
     "@nivo/stream": "^0.79.1",
     "@types/react-table": "^7.7.8",
+    "call-once": "^1.0.3",
```
What do these three packages do?

If they are for tests they can be placed in devDependencies.

alexmassen-hane replied:
Was a package for the trying to fix the unit tests. I have since removed it in the latest version, but the copy link button tests are failing :/

jdddog commented on theme/components/button.ts:
```diff
@@ -16,6 +16,14 @@
 const Button = {
   variants: {
+    shareButton: {
+      width: { base: "150px", ms: "100px", md: "150px" },
```
Missing the styling so that when you hover over it, the color changes.

jdddog reviewed (changes requested):
Thanks Alex, it is looking good. Most of my feedback is about fine tuning the styling now.

For the share button, on desktop it would be good to make it look the same as the Download / Wikipedia links.

On mobile:
- The logo / title and share button are not vertically centred because there is 12px of padding at the bottom of the logo / title. Remove the padding on mobile.
- On the smallest screens the title + share button can take up a bit too much space. On the base breakpoint, below sm 600px.
  - Reduce the size of the title.
  - Remove the share text from the share button.
    - Make sure to remove the extra right padding on the button icon in this case.

alexmassen-hane commented:
I've made the styling changes. The full link on the copy toast is quite a cluttered mess so I have neglected it. Please let me know if I've missed anything. 

We still need to investigate why the unit tests are failing for the "copy link button" tests.

![image](https://user-images.githubusercontent.com/104514872/199214930-6f89459f-6841-4d0c-b39b-810b89986388.png)

![image](https://user-images.githubusercontent.com/104514872/199214972-13275714-509c-4bc6-a9cb-20351903fc6a.png)

![image](https://user-images.githubusercontent.com/104514872/199215059-37d03a5a-427d-4512-bdba-d0ccef9bcf3d.png)

jdddog reviewed (approved):
Thanks for your work Alex, I think it's ready, so let's merge it.
